mobile/mobile-core/devices-service.ts

```diff
@@ -344,9 +344,20 @@ export class DevicesService extends EventEmitter implements Mobile.IDevicesServi
 	}
 
 	@exported("devicesService")
-	public deployOnDevices(deviceIdentifiers: string[], packageFile: string, packageName: string): Promise<void>[] {
+	public deployOnDevices(deviceIdentifiers: string[], packageFile: string, packageName: string | Mobile.IProjectIdentifier): Promise<void>[] {
 		this.$logger.trace(`Called deployOnDevices for identifiers ${deviceIdentifiers} for packageFile: ${packageFile}. packageName is ${packageName}.`);
-		return _.map(deviceIdentifiers, deviceIdentifier => this.deployOnDevice(deviceIdentifier, packageFile, packageName));
+		return _.map(deviceIdentifiers, async deviceIdentifier => {
+			const device = this.getDeviceByIdentifier(deviceIdentifier);
+
+			let identifier: string;
```

Comment:
the next few lines should be outside of the map, as they are not specific for device and there's no need to execute them each time

Reply:
I can't get the package name if I don't have the platform of the device.

Reply:
:man_facepalming: I'm stupid, ignore my comment